src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -670,6 +670,32 @@ private void FlushPendingOutput()
                         }
                     }
 
+                    //create the file in the main thread and flush the contents in the background thread
+                    // In case , there is an error in file creation, the logic would return from here and
+                    // not consume the exception
+                    try
+                    {
+                        string baseDirectory = Path.GetDirectoryName(transcript.Path);
+                        if(baseDirectory.Equals(transcript.Path.TrimEnd(System.IO.Path.DirectorySeparatorChar)))
+                        {
+                            Exception exception = PSTraceSource.NewArgumentException(transcript.Path,
+                            InternalHostUserInterfaceStrings.InvalidTranscriptFilePath, transcript.Path);
+                            throw exception;
+                        }
```

Comment:
Path.GetFileName("e:\testExistingFolder") returns testExistingFolder. 
Although in this case as well, the file generation will fail but the innerexception would be 
'Access to the path 'E:\scripts' is denied.'
Not sure if it gives the message that the failure is caused because file path resolve to a directory.
Also, this will create a difference in error for following scenarios :
Start-Transcript -Path 'C:\TestExistingFolder\' 
  inner exception : Cannot perform operation because the transcript file path is invalid.
Start-Transcript -Path 'C:\TestExistingFolder'
inner exception : Access to the path 'E:\scripts' is denied.